drivers/timer/Kconfig

```diff
@@ -64,6 +64,16 @@ config HPET_TIMER
 	  This option selects High Precision Event Timer (HPET) as a
 	  system timer.
 
+config HPET_TIMER_QUIRK_CNT_CLK_PERIOD_OFF_BY_1000
```

Comment:
Might suggest "HPET_CAP_PERIOD_UNIT_POW10" or something and set it to a default of 15.  That makes it sound more like the system vendor made a design decision and not just a mistake. :)

Also it avoids the ambiguity of what "off by 1000" means (x*1000?  x/1000?  x+1000?).

FWIW: I still maintain we want to remove the clock rate at runtime feature and just configure the clock rate at a platform level in DT or whatever.  There's non-trivial complexity in the kernel code due to this very driver, and this whole problem would go away.

Reply:
Though thinking about this now... powers of 10 are annoying to compute at compile time, so maybe specifying a unit in femptoseconds and setting the default to 1 is best?  Meh.

Reply:
Updated with a choice instead.